tests/run-make/optimization-remarks-dir-pgo/rmake.rs

```diff
@@ -0,0 +1,49 @@
+// This test checks the -Zremark-dir flag, which writes LLVM
+// optimization remarks to the YAML format. When using PGO (Profile
+// Guided Optimization), the Hotness attribute should be included in
+// the output remark files.
+// See https://github.com/rust-lang/rust/pull/114439
+
+//@ needs-profiler-support
+//@ ignore-cross-compile
+
+use run_make_support::{
+    has_extension, has_prefix, invalid_utf8_contains, llvm_profdata, run, rustc, shallow_find_files,
+};
+
+fn main() {
+    rustc().profile_generate("profdata").opt().input("foo.rs").output("foo").run();
+    run("foo");
+    llvm_profdata()
+        .merge()
+        .output("merged.profdata")
+        .input(
+            shallow_find_files("profdata", |path| {
+                has_prefix(path, "default") && has_extension(path, "profraw")
+            })
+            .get(0)
+            .unwrap(),
```

Comment:
Question/Suggestion: is the reason why we're doing shallow find here that the profdata filename isn't predictable? In any case, can you pull this out to a temp variable, because I feel like it's a bit nested and hard to follow.

```rs
let profdata_file = shallow_find_files("profdata", |path| {
        has_prefix(path, "default") && has_extension(path, "profraw")
    })
    .get(0)
    .unwrap();
```

Reply:
Indeed, it's a long sequence of numbers, and I previously used the hardcoded filename in a previous draft of this test, which caused failures on certain architectures.